controllers/authpolicy_controller.go

```diff
@@ -190,6 +198,24 @@ func (r *AuthPolicyReconciler) deleteNetworkResourceDirectBackReference(ctx cont
 	return r.DeleteTargetBackReference(ctx, targetNetworkObject, common.AuthPolicyBackRefAnnotation)
 }
 
+// reconcileRouteParentGatewayPolicies triggers the concurrent reconciliation of all policies that target gateways that are parents of a route
+func (r *AuthPolicyReconciler) reconcileRouteParentGatewayPolicies(ctx context.Context, route *gatewayapiv1beta1.HTTPRoute) error {
+	logger, err := logr.FromContext(ctx)
+	if err != nil {
+		return err
+	}
+	mapper := HTTPRouteParentRefsEventMapper{
+		Logger: logger,
+		Client: r.Client(),
+	}
+	requests := mapper.MapToAuthPolicy(route)
+	for i := range requests {
+		request := requests[i]
+		go r.Reconcile(context.Background(), request)
```

Comment:
This seems odd. Are we not dropping errors and reconcile requests being returned from this this way? Should it be done as part of a Watches in the `NewControllerManagedBy` ?

We have something "like this " 

```
Watches(&source.Kind{Type: &workv1.ManifestWork{}}, handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
			log.V(3).Info("enqueuing gateways based on manifest work change ", "work namespace", o.GetNamespace())
			requests := []reconcile.Request{}
			annotations := o.GetAnnotations()
			if annotations == nil {
				log.V(3).Info("no parent or anotations on manifest work ", "work ns", o.GetNamespace(), "name", o.GetName())
				return requests
			}
			key := annotations["kuadrant.io/parent"]
			ns, name, err := cache.SplitMetaNamespaceKey(key)
			if err != nil {
				log.Error(err, "failed to parse namespace and name from maifiest work")
				return requests
			}
			log.Info("requeuing gateway ", "namespace", ns, "name", name)
			requests = append(requests, reconcile.Request{
				NamespacedName: types.NamespacedName{Namespace: ns, Name: name},
			})
			return requests
		}), builder.OnlyMetadata).

```

Reply:
What kind of resource should we watch and map to AuthPolicies, @maleck13?

The intent here is, after reconciling an AuthPolicy attached to a HTTPRoute, to trigger the reconciliation of all AuthPolicies attached to Gateways that are parents of the HTTPRoute.

A few constraints are:
1. Neither HTTPRoute nor the Gateways are touched during any of those reconciliation loops. Only the internal custom resources might (i.e. `AuthorizationPolicy`, `AuthConfig`.)
2. We do not want to rely on back-ref annotations for this, as order matters here and therefore we want the latest state possible.

Reply:
Right. Prob wrong place to start figuring this out here. It would seem we need to trigger an event on the HTTPRoute (status update / annotation) or we need to track the parent policies. Do we have an issue for looking into a better option? I know it was discussed. IE have an in mem structure that we maintain of the policy hierarchy. Trigger a reconcile on parents on child change (or something along those lines) so I guess if we have an issue for that we could settle with this for now?

Reply:
I'll make sure to open one and add it here for ref.

Reply:
https://github.com/Kuadrant/architecture/issues/29